test/IdRegistry/IdRegistry.t.sol

```diff
@@ -300,6 +300,74 @@ contract IdRegistryTest is IdRegistryTestSuite {
         assertEq(idRegistry.getRecoveryOf(1), address(0));
     }
 
+    function testFuzzVerifyFidSignature(
+        address registrar,
+        uint256 recipientPk,
+        address recovery,
+        uint40 _deadline
+    ) public {
+        uint256 deadline = _boundDeadline(_deadline);
+        recipientPk = _boundPk(recipientPk);
+
+        address recipient = vm.addr(recipientPk);
+        bytes memory sig = _signRegister(recipientPk, recipient, recovery, deadline);
+
+        vm.prank(owner);
+        idRegistry.disableTrustedOnly();
+
+        assertEq(idRegistry.getIdCounter(), 0);
+        assertEq(idRegistry.idOf(recipient), 0);
+        assertEq(idRegistry.getRecoveryOf(1), address(0));
+
+        vm.expectEmit(true, true, true, true);
+        emit Register(recipient, 1, recovery);
+        vm.prank(registrar);
+        idRegistry.registerFor(recipient, recovery, deadline, sig);
+
+        assertEq(idRegistry.getIdCounter(), 1);
+        assertEq(idRegistry.idOf(recipient), 1);
+        assertEq(idRegistry.getRecoveryOf(1), recovery);
+
+        bytes32 digest = keccak256(bytes("message"));
+        bytes memory msg_sig = _signMsg(recipientPk, digest);
+        assertEq(idRegistry.verifyFidSignature(recipient, 1, digest, msg_sig), true);
+    }
+
+    function testFuzzCannotVerifyFidSignature(
```

Comment:
same comments about picking the right fuzz params

Reply:
Cleaned up the tests